workflow-templates/go-check.yml

```diff
@@ -6,8 +6,16 @@ jobs:
     name: Go checks
     steps:
       - uses: actions/checkout@v2
-      - name: golangci-lint
-        uses: golangci/golangci-lint-action@d9f0e73c0497685d68af8c58280f49fcaf0545ff # v2.5.1
+      - uses: actions/setup-go@v2
         with:
-          version: v1.33
-          args: "--disable-all --enable gofmt,govet"
+          go-version: "1.16.x"
+      - name: Install staticcheck
+        run: go install honnef.co/go/tools/cmd/staticcheck@be534f007836a777104a15f2456cd1fffd3ddee8 # v2020.2.2
+      - name: go vet
+        run: go vet ./...
+      - name: staticcheck
+        if: ${{ success() || failure() }}
+        run: |
+          set -o pipefail
+          staticcheck ./... | sed -e "s/\(.*\)\.go/.\/\1.go/g"
```

Comment:
perhaps document the sed with a TODO comment pointing at the setup-go issue.

Also, I think it would be clearer to use single quotes for regexps, so that shell escaping doesn't play a part. You can also use a different separator to not have to escape the slash:

```
sed -e 's@\(.*\)\.go@./\1.go/g'
```

I also wonder if you want `[^ ]+\.go` for filenames, as that might have fewer false positives like `foo bar.go`.

Reply:
I tried `[^ ]+\.go`, but sed doesn't like it.